src/PageParser.php

```diff
@@ -7,8 +7,13 @@
 use DOMXPath;
 
 class PageParser {
+
+	/** @var DOMXPath */
 	protected $xPath;
 
+	/** @var string[] Element ID values, used to fix duplicates. */
+	private static $ids = [];
```

Comment:
Looks good to me! Is there a reason why this array is static rather than dynamic?

Reply:
It's because we run PageParser on each separate chapter, but want a complete list of all IDs from all chapters, shared between all PageParser objects.

Reply:
I knew there was a strong reason for this, but couldn't figure out what it was. Thank you.